lib/data/external/AwsClient.js

```diff
@@ -24,15 +24,18 @@ class AwsClient {
     put(stream, size, keyContext, reqUids, callback) {
         const awsKey = this._createAwsKey(keyContext.bucketName,
            keyContext.objectKey, this._bucketMatch);
-        // TODO: if object to be encrypted, use encryption
-        // on AWS
-        return this._client.upload({
+        const uploadParams = {
             Bucket: this._awsBucketName,
             Key: awsKey,
             Body: stream,
             Metadata: keyContext.metaHeaders,
             ContentLength: size,
-        },
+        };
+        if (keyContext.cipherBundle) {
+            uploadParams.ServerSideEncryption = 'aws:kms';
+        }
+
```

Comment:
please add a test where you put an object to the aws backend in an encrypted bucket and then you get the object directly from aws to confirm that you can retrieve it and that it sends back the proper encryption headers.